ofrak_core/ofrak/core/magic.py

```diff
@@ -22,19 +28,47 @@ class Magic(ResourceAttributes):
     descriptor: str
 
 
+class _LibmagicDependency(ComponentExternalTool):
+    def __init__(self):
+        super().__init__(
+            "libmagic",
+            "https://www.darwinsys.com/file/",
+            install_check_arg="",
+            apt_package="libmagic1",
+            brew_package="libmagic",
+        )
+
+        try:
+            import magic as _magic
+
+            _LibmagicDependency._magic = _magic
+        except ImportError:
+            _LibmagicDependency._magic = None
+
+    async def is_tool_installed(self) -> bool:
+        return MAGIC_INSTALLED
+
+
+LIBMAGIC_DEP = _LibmagicDependency()
+
+
 class MagicAnalyzer(Analyzer[None, Magic]):
     """
     Analyze a binary blob to extract its mimetype and magic description.
     """
 
     targets = (File, GenericBinary)
     outputs = (Magic,)
+    external_dependencies = (LIBMAGIC_DEP,)
 
     async def analyze(self, resource: Resource, config=None) -> Magic:
         data = await resource.get_data()
-        magic_mime = magic.from_buffer(data, mime=True)
-        magic_description = magic.from_buffer(data)
-        return Magic(magic_mime, magic_description)
+        if not MAGIC_INSTALLED:
+            raise ImportError("libmagic does not seem to be installed!")
```

Comment:
Would it be better to check for (and handle) this error inside of [AbstractResource.run](https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_core/ofrak/component/abstract.py#L100), where other `ComponentMissingDependencyError`s are raised?